services/grpc/grpc.js

```diff
@@ -205,6 +207,45 @@ class GrpcService extends EventEmitter {
       }
     }
   }
+
+  /**
+   * Wait for a service to become active.
+   * @return {Promise<Object>} Object with `isActive` and `cancel` properties.
+   */
+  /**
+   * Wait for a service to become active.
+   * @param  {String} serviceName Name of service to wait for (Lightning, or WalletUnlocker)
+   * @return {Promise<Object>}     Object with `isActive` and `cancel` properties.
+   */
+  waitForService(serviceName) {
+    let successHandler
+    const activationEventName = `GRPC_${snakecase(serviceName).toUpperCase()}_SERVICE_ACTIVE`
+
+    /**
+     * Promise that resolves when service is active.
+     */
+    const isActive = new Promise(async resolve => {
+      // If the service is already active, return immediatly.
+      if (this.services[serviceName].is('connected')) {
```

Comment:
if we are requesting to wait for  a service that was not previously registered (and thus `this.services[serviceName]`) this will throw. is it expected?

Reply:
I think it probably should throw in this case. The alternative is that we resolve immediately in the case that the caller asks about an unknown service, but that seems wrong since the service isn't active.

Services get registered in the `init` method, so if the user was asking about an unregistered service it would imply that they had either not initialised the services first, or that they are simply asking about an unknown service - in either case, this is a user error and we should probably let it throw.

Reply:
Ok, let's leave it then